docs(TLS&TCP): update examples to reflect new port configuration and incl… (#8598)
Kong/docs.konghq.com: app/_src/kubernetes-ingress-controller/guides/services/tcp.md

1.  Configure TCPRoute (Gateway API Only)

    {:.note}

>     > If you are using the Gateway APIs (TCPRoute), your Gateway needs additional configuration under `listeners`.

Reviewer: ```suggestion
    > If you are using the Gateway APIs (`TCPRoute`), your Gateway needs additional configuration under `listeners`.
```

### Test the configuration

1. Check if the Service is ready on the route. Note that it may take a few moments for the Kong controller to process the route.
    {% capture the_code %}
{% navtabs codeblock %}
{% navtab Gateway API %}
```bash

> # Wait a few seconds for the controller to process the route
> sleep 5
> kubectl get tcproute echo-plaintext -ojsonpath='{.status.parents[0].conditions[?(@.reason=="Accepted")]}'

Reviewer: We can you `kubectl wait` here:

```suggestion
# Wait for the controller to process the route
kubectl wait tcproute echo-plaintext --for='jsonpath={.status.parents[0].conditions[?(@.reason=="Accepted")]}=True'
```

### Test the configuration

1. Check if the Service is ready on the route. Note that it may take a few moments for the Kong controller to process the route.
    {% capture the_code %}
{% navtabs codeblock %}
{% navtab Gateway API %}
```bash

> # Wait for the controller to process the route
> kubectl wait tcproute echo-plaintext --for='jsonpath={.status.parents[0].conditions[?(@.reason=="Accepted")]}=True'

Reviewer: ```suggestion
# Wait for the controller to process the route
kubectl wait tcproute echo-plaintext --for='jsonpath={.status.parents[0].conditions[?(@.type=="Accepted")].status}'=True
```

## Route based on SNI

{:.note}

> > For Gateway API, we use TLSRoute for SNI-based routing because it's specifically designed to handle TLS traffic with SNI information, while still proxying the underlying TCP stream. This is still considered TCP proxying, just with the added capability of routing based on the SNI in the TLS handshake.

Reviewer: ```suggestion
> For Gateway API, we use `TLSRoute` for SNI-based routing because it's specifically designed to handle TLS traffic with SNI information, while still proxying the underlying TCP stream. This is still considered TCP proxying, just with the added capability of routing based on the SNI in the TLS handshake.
```

{% include /md/kic/add-certificate.md hostname='tls9443.kong.example' release=page.release %}

> 1. Create the TLSRoute (for Gateway API) or TCPIngress (for Ingress) resource to route TLS-encrypted traffic to the `echo` service.

Reviewer: ```suggestion
1. Create the `TLSRoute` (for Gateway API) or `TCPIngress` (for `Ingress`) resource to route TLS-encrypted traffic to the `echo` service.
```

In real-world usage, you would create a DNS record for `tls9443.kong.example`pointing to your proxy Service's public IP address, which causes TLS clients to add SNI automatically. For this demo, add it manually using the OpenSSL CLI.

```bash
# If you haven't set the PROXY_IP variable yet
PROXY_IP=$(kubectl get svc -n kong kong-gateway-proxy -o jsonpath='{.status.loadBalancer.ingress[0].ip}')

> echo "hello" | openssl s_client -connect $PROXY_IP:9443 -servername tls9443.kong.example -quiet 2>/dev/null

Reviewer 2: The backendRef should use `1030` or the service's port that pointing to the `tlsecho` container's `1030` port. The `go-echo` image listens on:
- port 1025 for TCP (without SSL)
- port 1026 for UDP
- port 1027 for HTTP
- port 1030 for TLS.
I'll update the manifests here to introduce it with more details.

> echo "hello" | openssl s_client -connect $PROXY_IP:9443 -servername tls9443.kong.example -quiet 2>/dev/null

Reviewer: @tao12345666333 @randmonkey the service port listen above is still an issue

> echo "hello" | openssl s_client -connect $PROXY_IP:9443 -servername tls9443.kong.example -quiet 2>/dev/null

Reviewer 2: @pmalek updated the `example/echo-service.yaml` in assets. Please see if the service port is present now.
